app/api/api_v1/endpoints/lms_router.py

```diff
@@ -0,0 +1,45 @@
+from typing import List
+from pydantic import BaseModel
+from fastapi import APIRouter, Request, Depends, UploadFile, File
+from sqlalchemy.orm import Session
+from app.services import LmsSyncService
+from app.core.dependencies import (
+    get_db, PermissionDependency,
+    UserIsInstructorPermission
+)
+
+router = APIRouter()
+
+@router.get("/lms/downsync")
+async def downsync(
+    *,
+    db: Session = Depends(get_db),
+    perm: None = Depends(PermissionDependency(UserIsInstructorPermission))
+):
+    return await LmsSyncService(db).downsync()
+
+@router.get("/lms/downsync/students")
+async def downsync_students(
+    *,
+    db: Session = Depends(get_db),
+    perm: None = Depends(PermissionDependency(UserIsInstructorPermission))
+):
+    return await LmsSyncService(db).sync_students()
+
+@router.get("/lms/downsync/assignments")
+async def downsync_assignments(
+    *,
+    db: Session = Depends(get_db),
+    perm: None = Depends(PermissionDependency(UserIsInstructorPermission))
+):
+    return await LmsSyncService(db).sync_assignments()
+
+@router.post("/lms/grades")
+async def post_grades(
+    *,
+    db: Session = Depends(get_db),
+    perm: None = Depends(PermissionDependency(UserIsInstructorPermission)),
+    file: UploadFile = File(...)
```

Comment:
Honestly it may be easier here to just treat the grades CSV as a string parameter of the JSON body, I don't particularly care though. It should also take an assignment id parameter.

Reply:
Done - added assignment id as param
Though I'm leaving the csv as an uploadable file - we can revert if this turns out to be a hassle